resolver_conn_wrapper.go

```diff
@@ -99,33 +100,44 @@ func (ccr *ccResolverWrapper) resolveNow(o resolver.ResolveNowOption) {
 
 func (ccr *ccResolverWrapper) close() {
 	ccr.resolver.Close()
-	close(ccr.done)
+	atomic.StoreUint32(&ccr.done, 1)
+}
+
+func (ccr *ccResolverWrapper) isDone() bool {
+	return atomic.LoadUint32(&ccr.done) == 1
+}
+
+func (ccr *ccResolverWrapper) UpdateState(s resolver.State) {
+	if ccr.isDone() {
+		return
+	}
+	grpclog.Infof("ccResolverWrapper: sending update to cc: %v", s)
+	// TODO: add Channelz Trace Event
```

Comment:
The event in there is for the service config only.

I was thinking there probably should just be one event for the whole new `resolver.State`, regardless of the previous state - what do you think?

Reply:
I think it will be confusing for the people who read the trace events. They need to manually digest the trace to understand when the change in addresses or service config happens.I would prefer only logging the event when there's a change.

Reply:
If it's helpful for debugging, the event could also include the differences from the previous state - addresses added & removed, and old/new service config (if different).  That could go into one event or several - what do you think?

Reply:
Yes, I think combine them into one is ok as long as it's clear which change(s) at this update.